src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/BitSwapApiTest.cs

```diff
@@ -0,0 +1,528 @@
+                //Assert.True(wantTask.IsCanceled);
+            }
+            finally
+            {
+                await dfsService.StopAsync();
+            }
+        }
+        
+        [Fact]
+        public async Task Wants()
+        {
+            var fileSystem = Substitute.For<IFileSystem>();
+            fileSystem.GetCatalystDataDir().Returns(new DirectoryInfo(Path.Combine(Environment.CurrentDirectory,
+                $"dfs1-_{DateTime.Now:yyMMddHHmmssffff}")));
+            var dfsService = TestDfs.GetTestDfs(_testOutputHelper, fileSystem);
+            await dfsService.StartAsync();
+            try
+            {
+                var cts = new CancellationTokenSource();
+                var block = new DagNode(Encoding.UTF8.GetBytes("BitSwapApiTest unknown block"));
+                Task wantTask = dfsService.BitSwapApi.GetAsync(block.Id, cts.Token);
+
+                var endTime = DateTime.Now.AddSeconds(10);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                    {
+                        throw new Xunit.Sdk.XunitException("wanted block is missing");
+                    }
+                    
+                    await Task.Delay(100, cts.Token);
+                    var w = await dfsService.BitSwapApi.WantsAsync(cancel: cts.Token);
+                    if (w.Contains(block.Id))
+                    {
+                        break;
+                    }
+                }
+                
+                cts.Cancel();
+                var wants = await _dfsService.BitSwapApi.WantsAsync(cancel: cts.Token);
+                wants.ToArray().Should().NotContain(block.Id);
+
+                //Race condition between threads
+                //Assert.True(wantTask.IsCanceled);
+            }
+            finally
+            {
+                await _dfsService.StopAsync();
+            }
+        }
+
+        [Fact]
+        public async Task OnConnect_Sends_WantList()
+        {
+            _dfsService.Options.Discovery.DisableMdns = true;
+            _dfsService.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+            await _dfsService.StartAsync();
+
+            _dfsServiceOther.Options.Discovery.DisableMdns = true;
+            _dfsServiceOther.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+            await _dfsServiceOther.StartAsync();
+            
+            try
+            {
+                var local = _dfsService.LocalPeer;
+                var remote = _dfsServiceOther.LocalPeer;
+                _testOutputHelper.WriteLine($"this at {local.Addresses.First()}");
+                _testOutputHelper.WriteLine($"other at {remote.Addresses.First()}");
+
+                var data = Guid.NewGuid().ToByteArray();
+                var cid = new Cid
+                {
+                    Hash = MultiHash.ComputeHash(data)
+                };
+                
+                var _ = _dfsService.BlockApi.GetAsync(cid);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)